### ISSUE TITLE
MacOS Monterey 12.3 deletes Python2 hence nodeenv fails

### ISSUE THREAD

**Issue by slimandslam** [ekalinin/nodeenv#301]
Just as the subject line says:

```
$ python
-bash: python: command not found
$ nodeenv
env: python: No such file or directory
$
```

Of course, python3 exists. But.....

```
$ sudo ln -s /usr/bin/python3 /usr/bin/python
Password:
ln: /usr/bin/python: Operation not permitted
$
```

I guess you have to disable SIP to make this symlink:  https://developer.apple.com/documentation/security/disabling_and_enabling_system_integrity_protection

I guess the question is: does nodeenv work properly with python 3  ?

Clarification: MacOS 12.3 deletes Python2 and /usr/bin/python (which breaks anything with #!/usr/bin/python in it)

**Comment by slimandslam** [comment]
I guess the question is: does nodeenv work properly with Python3 ?

**Comment by maxmelamed** [comment]
I've been using a patched version of nodeenv locally that uses `python3` in the shebang (exactly like https://github.com/ekalinin/nodeenv/pull/305) and haven't seen any issues.

**Comment by ekalinin** [comment]
Yeah, I believe this patch is working well for the systems where python3 is installed. But at the same time I think it will break ones where python3 is not installed.

Thanks for the patch, btw!

**Comment by maxmelamed** [comment]
> Yeah, I believe this patch is working well for the systems where python3 is installed. But at the same time I think it will break ones where python3 is not installed.

Right, definitely don't want to break any existing users. If the `python_requires` gets updated accordingly in `setup.py` though, `easy_install` and `pip install` would know that the newer versions are incompatible with py2 and wouldn't update unless users are on python3 already (this would prevent them from getting future updates unless they install python3, however).

**Comment by ekalinin** [comment]
nodeenv is effectively Python 3 only now - tox and CI run 3.10-3.14, and the `nodeenv` console script takes its shebang from the interpreter you install it with. Installing under python3 (`python3 -m pip install nodeenv`) avoids the reported `env: python: No such file or directory`.

The stale `python_requires = >=2.7` left in `setup.py` is a separate metadata cleanup, best done together with the packaging work in #374.

Closing as obsolete.
